ferrocene/tools/self-test/src/env.rs

```diff
@@ -18,18 +20,55 @@ pub(crate) const SELFTEST_RUST_HASH: Option<&str> = option_env!("SELFTEST_RUST_H
 pub(crate) const SELFTEST_CARGO_HASH: Option<&str> = option_env!("SELFTEST_CARGO_HASH");
 
 /// Run-time environment variables
+#[derive(Debug, Default)]
 pub(crate) struct Env {
     /// `PATH` environment variable.
     ///
     /// Specifies a set of directories where executable programs are located.
     ///
     /// Usually this variable is present in most operating systems and does not
     /// need to be set explicitly.
-    pub(crate) path: Option<OsString>,
+    pub(crate) path: OsString,
+    pub(crate) print_detailed_args: Option<String>,
+    pub(crate) print_detailed_errors: Option<String>,
 }
 
 impl Env {
-    pub(crate) fn gather() -> Self {
-        Self { path: std::env::var_os("PATH") }
+    pub(crate) fn gather() -> Result<Self, Error> {
+        Self::new(
+            std::env::var_os("PATH"),
+            std::env::var("FST_PRINT_DETAILED_ARGS").ok(),
+            std::env::var("FST_PRINT_DETAILED_ERRORS").ok(),
```

Comment:
We should just store a boolean of whether the variable is present or not, as that's all the only way we are using those variables.

Reply:
Done